source/process/regex-library-draft.md

```diff
@@ -0,0 +1,175 @@
+* **no space open parens**: Find and replace an opening parenthesis with no space before<br>F: `(\w</span>)(\()`<br>R: `\1 \2`
+* **begin span spacing**: Find spans lacking a space before<br> F: `([a-z]+)(<span)`<br>R: `\1 \2`
+* **space after first tag**: Find and replace opening tags with a space after<br>F: `<([^>])> (.*?)`<br>R: `<\1>\2`
+* **space before last tag**: Find and replace closing tags with a space before<br>F: `</(p|td|h1|h2|h3)>`<br>R: `</\1>`
+* **dash spacing**: Find dashes with potential spacing issues<br>F: `(\s[^>/= ]*\s[-–][^</= ]*\s|\s[^>/= ]*[-–]\s[^</= ]*\s)`
+* **space after comma**: Find a comma with no space after<br>F: `,([^"’”'<0-9 —\)]+)<br>R: , \1`
+
+</details>
+
+<details close>
+
+<summary>Spans</summary>
+
+* **span combine (1)**: In this Regex Library navigate to _Clean and Code > Spacing > **no space between spans**_ and check before running span combine. Find and replace to combine the content of spans with the same class<br>F: `<span class="([^"]*)">([^<]*)</span>(\s*)<span class="\1">([^<]*)</span>`<br>R: `<span class="\1">\2\3\4</span>`
+* **span combine (2)**: Find and replace spans that can be combined into a single class<br>F: `<span class="([^"]*)"><span class="([^"]*)">([^<]*)</span></span>`<br>R: `<span class="\1 \2">\3</span>`
+* **remove spans from headings**: Find spans in headings that are potentially not needed<br>F: `(<h\d[^>]*>.*?)<span(\s*class="(?!label)[^"]*")*>([^<]*)</span>(.*?</h\d>)`<br>R: `\1\3\4`
+* **remove space within spans**: Find spans with a space inside<br>F: `<span class="([^"]+)"> ([^<]+)</span>`<br>R: `<span class="\1">\2</span>` (include the space _before_ the span)<br><br>F: `<span class="([^"]+)">([^<]+) </span>`<br>R: `<span class="\1">\2</span>` (include the space _after_ the span)
+* **move non-english chars in span**: Find and replace the class of a span containing non-english characters<br>F: `<span class="(italic|i)">([^a-zA-Z0-9\s]+)</span>`<br>R: `<span class="\1">\2</span>`
+* **remove unnecessary span**: Find spans around punctuation and replace without the span<br>F: `<span class="[^"]*">(‘|“|’|”|\.|\)|\(|\?|!|,)+</span>`<br>R: `\1`
+* **repeating spans**: Find and replace adjacent spans that repeat<br>F: `<span class="([^\n<>]+)">([^\n<>]+)</span><span class="\1">`<br>R: `<span class="\1">\2`
+
+</details></blockquote>
+
+</details>
+
+<br>
+
+<details close>
+
+<summary>Enhance</summary><blockquote>
+
+<details close>
+
+<summary>Abbreviations</summary>
+
+* **tables to ABBR 1**: convert tables to abbreviation lists<br>F: `<tr>\s*<td>(.*?)</td>\s*<td>(.*?)</td>\s*</tr>`<br>R: `<dt epub:type="glossterm"><dfn>\1</dfn></dt><dd epub:type="glossdef">\2</dd>`
+* **tables to ABBR 2**: after running tables to ABBR 1 use this regex to format the lists new lines<br>F: `<dfn>(.*?)</dfn></dt><dd epub:type="glossdef">(.*?)</dd>`<br>R: `\n\s\s\s\s\s\s\s\s\s\s\s\s<dfn>\1</dfn>\n\s\s\s\s\s\s\s\s\s\s</dt>\n\s\s\s\s\s\s\s\s\s\s<dd epub:type="glossdef">\2</dd>`
```

Comment:
I think instead of using repeated `\s` here for spacing in the replacement, we'll need to use actual spaces.
```
\n            <dfn>\1</dfn>\n          </dt>\n          <dd epub:type="glossdef">\2</dd>
```
As far as I know, using `\s` in the replacement will insert an actual 's' character